cmd/main/main.go

```diff
@@ -5,13 +5,17 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"github.com/iamananya/ginco-task/pkg/middlewares"
 	"github.com/iamananya/ginco-task/pkg/routes"
 	_ "github.com/jinzhu/gorm/dialects/mysql"
 )
 
 func main() {
 	r := mux.NewRouter()
 	routes.RegisterUserRoute(r)
-	http.Handle("/", r)
-	log.Fatal(http.ListenAndServe("localhost:9010", r))
+	// Apply authentication middleware to all routes
+	authenticatedRouter := middlewares.AuthenticationMiddleware(r.ServeHTTP)
+
+	http.Handle("/", authenticatedRouter)
```

Comment:
What does the / (root) API return?

Reply:
It returns everything after the / sign. I will remove it as it is redundant.